### PR TITLE
Add statement descriptor test for renewal order

### PR DESCRIPTION
**Pull request by v18**
Based on @RadoslavGeorgiev's change [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/963#discussion_r316327171), I realized we could actually test the statement descriptor 🎉

 🙏 thanks Rado :)

The test works by adding the `woocommerce_stripe_settings` option to the test DB, and setting the payment method to stripe.

This could either be merged into `add/off-session-for-recurring-payments-938` before #955 is merged, or directly into `sca-subs/master` after #955 is merged.

### CODE REVIEW

File: tests/phpunit/test-wc-stripe-sub-renewal.php [hunk 1]
```diff
@@ -39,6 +46,23 @@ public function setUp() {
...
+		);
+	}
+
+	/**
+	 * Tears down the stuff we set up.
+	 */
+	public function tearDown() {
+		delete_option( 'woocommerce_stripe_settings' );
+
+		parent::tearDown();
```
Comment (RadoslavGeorgiev) [inline tests/phpunit/test-wc-stripe-sub-renewal.php]: Small nitpick: Unless there is a need to prepare data for the parent constructor, let's call it in the beginning of the method.

Comment (v18) [reply tests/phpunit/test-wc-stripe-sub-renewal.php]: There isn't a need per se, it's just that if the tests fail for any reason, we don't have to write `catch` code in order to make sure that we `delete_option` at the end of the test run.

**Comment by v18** [comment]
The test for WC 2.6.14 was failing, I fixed it by correctly setting the order's payment method using this:

```
$renewal_order->payment_method = 'stripe';
```

File: tests/phpunit/test-wc-stripe-sub-renewal.php (hunk 1 above)
Comment (RadoslavGeorgiev) [reply tests/phpunit/test-wc-stripe-sub-renewal.php]: Isn't `tearDown` being called irregardless of whether the tests fail or succeed?

Comment (v18) [reply tests/phpunit/test-wc-stripe-sub-renewal.php]: Ah, I gotcha now 👌 Done.

**Pull request merged**